sirepo/uri_router.py

```diff
@@ -185,7 +185,7 @@ def uri_for_api(api_name, params=None, external=True):
         str: formmatted external URI
     """
     r = _api_to_route[api_name]
-    res = (sirepo.uri.server_root(external=external) + r.base_uri).rstrip('/')
+    res = (flask.url_for('_dispatch_empty', _external=external) + r.base_uri).rstrip('/')
```

Comment:
This will not work if external is true, it will strip too many slashes with `app_root(sim_type=None)`. This is unusual, but easily fixable:

```py
res = r.base_uri
```

`res` doesn't have any slashes at this point.

Then at the end:

```py
return flask.url_for('_dispatch_empty', _external=external) + res
```

The code was broken for this case before. It would be good to have a test. Add this case to uri_router_test.py test_uri_for_api:

```py
        pkeq('/', uri_router.uri_for_api('root', params={}, external=False))
```

It will have very confusing output so run it this way:

```sh
pykern test uri_router_test.py case=uri_for_api
```

This will still be confusing, because it has to run the test case in the server, but look for the PKFail:

```txt
pykern.pkunit.PKFail: uri_router_test.py:64:t expect=/ != actual=
```

Reply:
ad61fc813610afe4d4c57cbe0e4322c503d6ac6c